src/JacobianByHand.jl

```diff
@@ -1,30 +1,33 @@
 """
 First step in camera projection
 """
-function P1(r, t, X)
+function P1!(r, t, X, P1_vec, P1_cross)
   θ = norm(r)
-  k = r / θ
-  return cos(θ) * X + sin(θ) * cross(k, X) + (1 - cos(θ)) * dot(k, X) * k + t
+  P1_vec .= r ./ θ
+  P1_cross[1] = P1_vec[2]*X[3]-P1_vec[3]*X[2]
+  P1_cross[2] = P1_vec[3]*X[1]-P1_vec[1]*X[3]
+  P1_cross[3] = P1_vec[1]*X[2]-P1_vec[2]*X[1]
```

Comment:
there is a cross product function

Reply:
``cross!`` function used in the last commit